proxy/http2/Http2ConnectionState.cc

```diff
@@ -1584,7 +1584,7 @@ Http2ConnectionState::send_headers_frame(Http2Stream *stream)
   if (header_blocks_size <= static_cast<uint32_t>(BUFFER_SIZE_FOR_INDEX(buffer_size_index[HTTP2_FRAME_TYPE_HEADERS]))) {
     payload_length = header_blocks_size;
     flags |= HTTP2_FLAGS_HEADERS_END_HEADERS;
-    if (h2_hdr.presence(MIME_PRESENCE_CONTENT_LENGTH) && h2_hdr.get_content_length() == 0) {
+    if ((h2_hdr.presence(MIME_PRESENCE_CONTENT_LENGTH) && h2_hdr.get_content_length() == 0) || stream->is_body_done()) {
```

Comment:
`Http2Stream::is_body_done()` returns `true` when `Http2Stream::write_vio.ntodo()` is 0.
If this approach goes well, this should be renamed to avoid confusion.